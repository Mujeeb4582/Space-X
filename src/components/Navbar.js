import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../img/logo.png';

const Navbar = () => {
  const activeStyle = ({ isActive }) => ({
    textDecoration: isActive ? 'underline' : 'none',
    fontWeight: 600,
    color: '#2088ff',
  });

  return (
    <header className="md:h-16 flex flex-col md:flex-row justify-between items-center my-6 mx-12 border-b-2 border-black pb-4">
      <NavLink
        className="h-full flex flex-row justify-between items-center mb-4"
        to="/"
      >
        <img className="h-10 mr-4 lg:h-14" src={logo} alt="logo" />
        <span className="text-2xl md:text-3xl">
          Space Travelers&apos; Hub
        </span>
      </NavLink>
      <nav className="flex gap-3 items-center">
        <NavLink
          to="/"
          style={activeStyle}
        >
          <span className="">Rockets</span>
        </NavLink>
        <NavLink
          to="/missions"
          style={activeStyle}
        >
          <span className="">Missions</span>
        </NavLink>
        <div className="h-4 border-r-2 border-black" />
        <NavLink
          to="/my-profile"
          style={activeStyle}
        >
          <span className="">My Profile</span>
        </NavLink>
      </nav>
    </header>
  );
};
export default Navbar;
