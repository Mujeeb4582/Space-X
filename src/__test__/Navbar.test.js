// file for navbar test
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Test Navbar component', () => {
  it('should renders Navbar component', () => {
    const NavbarComponent = TestRenderer.create(<Router><Navbar /></Router>).toJSON();
    expect(NavbarComponent).toMatchSnapshot();
  });
});
