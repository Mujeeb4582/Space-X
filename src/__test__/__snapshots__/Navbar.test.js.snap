// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar component should renders Navbar component 1`] = `
<header
  className="md:h-16 flex flex-col md:flex-row justify-between items-center my-6 mx-12 border-b-2 border-black pb-4"
>
  <a
    aria-current="page"
    className="h-full flex flex-row justify-between items-center mb-4 active"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      className="h-10 mr-4 lg:h-14"
      src="logo.png"
    />
    <span
      className="text-2xl md:text-3xl"
    >
      Space Travelers' Hub
    </span>
  </a>
  <nav
    className="flex gap-3 items-center"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "#2088ff",
          "fontWeight": 600,
          "textDecoration": "underline",
        }
      }
    >
      <span
        className=""
      >
        Rockets
      </span>
    </a>
    <a
      className=""
      href="/missions"
      onClick={[Function]}
      style={
        Object {
          "color": "#2088ff",
          "fontWeight": 600,
          "textDecoration": "none",
        }
      }
    >
      <span
        className=""
      >
        Missions
      </span>
    </a>
    <div
      className="h-4 border-r-2 border-black"
    />
    <a
      className=""
      href="/my-profile"
      onClick={[Function]}
      style={
        Object {
          "color": "#2088ff",
          "fontWeight": 600,
          "textDecoration": "none",
        }
      }
    >
      <span
        className=""
      >
        My Profile
      </span>
    </a>
  </nav>
</header>
`;
